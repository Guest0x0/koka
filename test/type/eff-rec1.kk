div effect cow
  ctl moo() : (() -> <cow,div> ())


val cow = handler
  ctl moo()
    println("moo")
    resume( { moo()() } )

fun main()
  with cow
  moo()()
