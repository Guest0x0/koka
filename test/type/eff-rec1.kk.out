type/eff-rec1/cow: forall<a,e> (() -> <console,cow|e> a) -> <console|e> a
type/eff-rec1/main: () -> <console,div> ()
type/eff-rec1/moo: () -> <cow,div> (() -> <cow,div> ())